### PR TITLE
Backoffice `Table` cell

### LINKED ISSUE #1349: Backoffice Table cell

### PR DESCRIPTION
**Pull request by Omri-Levy**
Added a new component to the backoffice, added it to the cells map, and made sure it follows a similar interface to the rest of the existing cells.

### CODE REVIEW

File: apps/backoffice-v2/src/common/components/atoms/Table/Table.tsx
Comment (Omri-Levy) [inline apps/backoffice-v2/src/common/components/atoms/Table/Table.tsx:1]: Importing the existing table components from `@ballerine/ui` breaks Storybook. Storybook made this PR go significantly faster.

File: apps/backoffice-v2/package.json [hunk 1]
```diff
@@ -68,6 +68,7 @@
     "@radix-ui/react-separator": "^1.0.2",
     "@radix-ui/react-slot": "^1.0.1",
     "@tanstack/react-query": "^4.19.1",
+    "@tanstack/react-table": "^8.9.2",
```
Comment (Omri-Levy) [inline apps/backoffice-v2/package.json:71]: Using `react-table` for the `Table` cell both makes adding sorting and such trivial, and their docs has a full snippet I could copy-paste for rendering.

File: apps/backoffice-v2/src/pages/Entity/components/TableCell/TableCell.stories.tsx
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Entity/components/TableCell/TableCell.stories.tsx:1]: Also covers empty columns or data.

File: apps/backoffice-v2/src/pages/Entity/components/TableCell/interfaces.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  TableCaption,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from '../../../../common/components/atoms/Table';
+import { ColumnDef, RowData, TableOptions } from '@tanstack/react-table';
+
+export interface ITableCellProps<TData extends RowData, TValue = unknown> {
+  // Props to be used explicitly by the cell inside 'value'. Props outside 'value' to be used by the blocks API.
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/pages/Entity/components/TableCell/interfaces.ts:14]: Do you feel like the comments in this file are necessary for the understanding of the interface?

File: apps/backoffice-v2/src/pages/Entity/components/TableCell/TableCell.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+              </TableRow>
+            ))}
+          {!table.getRowModel().rows?.length && (
+            <TableRow {...value?.props?.row}>
+              <TableCellUI
+                colSpan={value?.columns?.length}
+                className="h-24 text-center"
+                {...value?.props?.cell}
+              >
+                No results.
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/pages/Entity/components/TableCell/TableCell.tsx:95]: Did you get from Nitzan a design for an empty table?

File: apps/backoffice-v2/src/pages/Entity/components/TableCell/TableCell.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
+import { flexRender, getCoreRowModel, RowData, useReactTable } from '@tanstack/react-table';
+import { ElementRef, ForwardedRef, forwardRef } from 'react';
+import {
+  Table,
+  TableBody,
+  TableCaption,
+  TableCell as TableCellUI,
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/pages/Entity/components/TableCell/TableCell.tsx]: How about `TableCellComponent`?

**Review by tomer-shvadron — APPROVED** [review]
Left a couple of small comments. Awesome work!

File: apps/backoffice-v2/src/pages/Entity/components/TableCell/interfaces.ts (hunk 2 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/pages/Entity/components/TableCell/interfaces.ts:14]: My thought process here is wondering how this interface would look like to someone who is unfamiliar with blocks and their concepts. Let me know if they should still be removed with that in mind.

File: apps/backoffice-v2/src/pages/Entity/components/TableCell/TableCell.tsx (hunk 3 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/pages/Entity/components/TableCell/TableCell.tsx:95]: We're in the process of matching the design to Figma. I made sure he is aware of the various states.

File: apps/backoffice-v2/src/pages/Entity/components/TableCell/TableCell.tsx (hunk 4 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/pages/Entity/components/TableCell/TableCell.tsx]: Good idea. I'm on it. 👍🏼

File: apps/backoffice-v2/src/pages/Entity/components/TableCell/interfaces.ts (hunk 2 above)
Comment (tomer-shvadron) [reply apps/backoffice-v2/src/pages/Entity/components/TableCell/interfaces.ts:14]: I think this type of instructions could go really well into a `README.md` file as they are relevant for a lot of our UI components, making a good reference for new developers

File: apps/backoffice-v2/src/common/components/molecules/JsonDialog/JsonDialog.tsx
Comment (Omri-Levy) [inline apps/backoffice-v2/src/common/components/molecules/JsonDialog/JsonDialog.tsx:1]: Same issue as `Table` where importing from `@ballerine/ui` breaks Storybook.

**Pull request merged**